core/win32/drwinapi/ntdll_redir.c

```diff
@@ -299,26 +299,26 @@ RtlAllocateHeap(HANDLE heap, ULONG flags, SIZE_T size);
 void *
 wrapped_dr_alloc(ULONG flags, SIZE_T size)
 {
-    byte *mem;
-    ASSERT(sizeof(size_t) >= HEAP_ALIGNMENT);
-    size += sizeof(size_t);
-    mem = global_heap_alloc(size HEAPACCT(ACCT_LIBDUP));
+    /* HeapAlloc returns 16-byte-aligned for 64-bit and 8-byte-aligned for 32-bit.
```

Comment:
Just for my understanding, why wasn't redirect_malloc called prior?

Reply:
Prior to this PR redirect_malloc was only used on UNIX, and a parallel implementation of adding a size field was used here on Windows, just for historical reasons.  Somehow we never saw any problems from the smaller alignment.  Now that it's more complex to add a layer around DR's allocator I made them share the same impl in redirect_malloc.

Reply:
"just for historical reasons": I think they each evolved separately and just nobody ever took a step to share the code (it was pretty simple code before).